src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1079 @@
+        if (newPan > mPanMax)
+        {
+            newPan = mPanMax;
+        }
+        if (newPan < mPanMin)
+        {
+            newPan = mPanMin;
+        }
+
+        hasAtLeastOneValue = true;
+    }
+
+    if (tiltDelta.HasValue())
+    {
+        if (!HasFeature(Feature::kMechanicalTilt))
+        {
+            ChipLogError(Zcl, "Mechanical Tilt not supported although tiltDelta value provided");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+            return;
+        }
+        int16_t tiltDeltaValue = tiltDelta.Value();
+        if (tiltDeltaValue > (mTiltMax - mTiltMin) || tiltDeltaValue < -(mTiltMax - mTiltMin))
+        {
+            ChipLogError(Zcl, "TiltDelta value received is out of range.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+
+        newTilt = static_cast<int16_t>(newTilt + tiltDeltaValue);
+        if (newTilt > mTiltMax)
+        {
+            newTilt = mTiltMax;
+        }
+        if (newTilt < mTiltMin)
+        {
+            newTilt = mTiltMin;
+        }
+
+        hasAtLeastOneValue = true;
+    }
+
+    if (zoomDelta.HasValue())
+    {
+        if (!HasFeature(Feature::kMechanicalZoom))
+        {
+            ChipLogError(Zcl, "Mechanical Zoom not supported although zoomDelta value provided");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+            return;
+        }
+        int8_t zoomDeltaValue = zoomDelta.Value();
+        if (zoomDeltaValue > (mZoomMax - 1) || zoomDeltaValue < -(mZoomMax - 1))
+        {
+            ChipLogError(Zcl, "ZoomDelta value received is out of range.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+
+        newZoom = static_cast<int8_t>(newZoom + zoomDeltaValue);
+        if (newZoom > mZoomMax)
+        {
+            newZoom = static_cast<int8_t>(mZoomMax);
+        }
+        if (newZoom < 1)
+        {
+            newZoom = 1;
+        }
+
+        hasAtLeastOneValue = true;
+    }
+
+    // Was a value received in the command
+    if (!hasAtLeastOneValue)
+    {
+        ChipLogError(Zcl, "MPTZRelativeMove contains no actionable fields");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+        return;
+    }
+
+    // Check with the delegate that we're in a position to change any of the PTZ values
+    //
+    if (!mDelegate->CanChangeMPTZ())
+    {
+        ChipLogDetail(Zcl, "Device not able to process MPTZ relative value change");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Busy);
+        return;
+    }
+
+    // Call the delegate to simply set the newly calculated MPTZ values based on the deltas received
+    //
+    status = mDelegate->MPTZRelativeMove(Optional(static_cast<int16_t>(newPan)), Optional(static_cast<int16_t>(newTilt)),
```

Comment:
Why do you need those casts?  newPan and newTilt are already those types...  Just `MakeOptional(newPan)` etc.